src/include/mpir_err.h

```diff
@@ -810,28 +810,38 @@ void MPIR_Handle_fatal_error(struct MPIR_Comm *comm_ptr, const char fcname[], in
 #define MPIR_ERR_SETANDSTMT(err_,class_,stmt_,msg_)                     \
     do {                                                                \
         err_ = MPIR_Err_create_code(err_,MPIR_ERR_RECOVERABLE,__func__,   \
-                                    __LINE__, class_, msg_, 0); stmt_ ; \
+                                    __LINE__, class_, msg_, 0);         \
+        assert(err_);                                                   \
+        stmt_ ;                                                         \
```

Comment:
I had added it in a previous version, but I removed it because the assert is generally useful even if the warning wasn't there.  It's an extra level of error checking to make sure someone doesn't accidentally use the error creation function incorrectly (e.g., by setting both the error code and error class as `MPI_SUCCESS`).

Reply:
OK.